core/src/main/scala/kafka/controller/TopicDeletionManager.scala

```diff
@@ -88,6 +86,10 @@ class TopicDeletionManager(controller: KafkaController,
   val deleteTopicStateChanged: AtomicBoolean = new AtomicBoolean(false)
   var deleteTopicsThread: DeleteTopicsThread = null
   val isDeleteTopicEnabled = controller.config.deleteTopicEnable
+  val topicsToBeDeleted: mutable.Set[String] = mutable.Set.empty[String]
+  if (isDeleteTopicEnabled)
+    topicsToBeDeleted ++= initialTopicsToBeDeleted
+  val partitionsToBeDeleted: mutable.Set[TopicAndPartition] = topicsToBeDeleted.flatMap(controllerContext.partitionsForTopic)
```

Comment:
@MayureshGharat : Suppose that someone deleted a topic when all brokers are down. Then we started all brokers. The controller won't get a topic deletion event since the event happened when the broker was down. In this case, we still need the logic to delete what's in deleteTopic path.